### PR TITLE
Fix creation of new mods

### PR DESCRIPTION
**Pull request by DasSkelett**
## Problem
#264 surfaced a bug which makes creating new mods broken now.

Until #264, we primarily filled the **game version** selector in `create.html` during (server-side) template rendering.
This step assigns `game_version.friendly_version` to the `value` attribute:
https://github.com/KSP-SpaceDock/SpaceDock/blob/e91894465c3a2885531355b9c813493ad93eaca4/templates/create.html#L72-L76

As soon as you switch the selected **game**, some JavaScript code is executed to refill the **game version** selector. However this JS code assigns `game_version.id` to the `value` attribute:
https://github.com/KSP-SpaceDock/SpaceDock/blob/e91894465c3a2885531355b9c813493ad93eaca4/templates/create.html#L179-L181

Since #264, the JS code is also executed once after page load, which results in the id being assigned all the time. The value of the `value` attribute is sent in the XHR request then.

However, the API expects `game_version.friendly_version`. So while until #264 creating mods was only broken when you changed the selected game once (which isn't possible on prod. because we only have one), now it's broken all the time.

## Changes
Now the API expects `game_version.id` in `create_mod()`, and also no longer needs a separate `game-id` form parameter, since we can get it from the `game_version`.

I also changed `update_mod()` and `update.html` to also use the `id` to keep it the same.

I removed some code from `create.html` which I missed to delete in #264. No need to fill the `game_version` selector if we overwrite it in JS anyways.

---

I had the choice between using the `id` everywhere or the `friendly_version`. I went with the `id` because it's always unique and you can derive the game from it.
But I'm also open to use `friendly_version` instead. Feel free to discuss what you think should be used.

A second commit is coming up which moves the embedded JS part into `update.coffee`.

### CODE REVIEW

**Comment by HebaruSan** [comment]
#227 seems really closely related. Maybe we can address that as long as we're working with all this code.

**Comment by HebaruSan** [comment]
> I went with the `id` because it's always unique and you can derive the game from it.
> But I'm also open to use `friendly_version` instead. Feel free to discuss what you think should be used.

I would lean towards keeping the API the same in case anyone else is using it. Changing an API properly is a tricky business, usually requires version specifiers so clients know what format to use, etc. Or at least if we want to support the internal id, put it in a parameter with a new name (kind of like @allista's suggestion in #227).

File: KerbalStuff/blueprints/api.py [hunk 1]
```diff
@@ -583,19 +580,10 @@ def create_mod():
...
-        return {'error': True, 'reason': 'Game does not exist.'}, 400
-    game_version_id = db.query(GameVersion.id) \
-        .filter(GameVersion.game_id == game.id) \
-        .filter(GameVersion.friendly_version == game_version) \
-        .first()
-    if not game_version_id:
+    game_version = GameVersion.query.get(game_version_id)
+    if not game_version:
         return {'error': True, 'reason': 'Game version does not exist.'}, 400
+    game = game_version.game
```
Comment (HebaruSan) [inline KerbalStuff/blueprints/api.py:586]: Should we check `game.active` here? It's odd that we have the ability to deactivate a game but then we allow creation of new mods for it.

https://github.com/KSP-SpaceDock/SpaceDock/blob/b7cb750b5b71e497f5d791c1754e55c1ac90ed0c/KerbalStuff/objects.py#L122-L126

**Comment by DasSkelett** [comment]
Second to latest commit moves JS to CoffeScript, latest commit moves from using `GameVersion.id` to `Gameversion.friendly_version` and brings back the `game-id`, and the `game-short`, and the deprecated `game` parameter in `create_mod()`.

File: frontend/coffee/create.coffee [hunk 2]
```diff
@@ -111,3 +111,70 @@ window.addEventListener('drop', (e) ->
 
 document.getElementById('submit').removeAttribute('disabled')
 $('[data-toggle="tooltip"]').tooltip()
+
+
+updategame = ->
+    gamename = $("#mod-game option:selected").text()
+    slug = $("#mod-game option:selected").attr("data-slug")
```
Comment (HebaruSan) [inline frontend/coffee/create.coffee:118]:
```suggestion
```

Are these old variables used for anything?

File: frontend/coffee/create.coffee [hunk 3]
```diff
@@ -111,3 +111,70 @@ window.addEventListener('drop', (e) ->
...
+    gid = $("#mod-game option:selected").attr("value")
+    # TODO: don't hardcode the production game id here.
+    #       Do we have a game.ckan_enabled property?
+    if (gid != "3102")
```
Comment (HebaruSan) [inline frontend/coffee/create.coffee:122]:
```suggestion
    if gid != "3102"
```

`if` doesn't require parentheses in Coffeescript.

File: frontend/coffee/create.coffee [hunk 4]
```diff
@@ -111,3 +111,70 @@ window.addEventListener('drop', (e) ->
...
+                $('<option value="' + data.friendly_version + '">' + data.friendly_version + '</option>').appendTo("#mod-game-version")
+            )
+            $("#mod-game-version").trigger("chosen:updated")
+
+    )
+
+
+# Check if there's a game preselected, if yes, get the game versions for it.
+preselected = $("#mod-game option:selected").attr("value")
+if (preselected != null)
```
Comment (HebaruSan) [inline frontend/coffee/create.coffee:145]:
```suggestion
if preselected != null
```

File: frontend/coffee/create.coffee [hunk 5]
```diff
@@ -111,3 +111,70 @@ window.addEventListener('drop', (e) ->
...
+    no_results_text: "No Options found",
+    width: '100%'
+)
+$("#mod-game").chosen({width: '100%'}).change(() ->
+    updategame()
+    updategameversions($(this).val())
+)
+
+licsel = $("#mod-license option:selected").html()
+if (licsel == "Other")
```
Comment (HebaruSan) [inline frontend/coffee/create.coffee:170]:
```suggestion
if licsel == "Other"
```

File: frontend/coffee/create.coffee [hunk 6]
```diff
@@ -111,3 +111,70 @@ window.addEventListener('drop', (e) ->
...
+    $("#mod-other-license").removeClass("hidden").show()
+else
+    $("#mod-other-license").addClass("hidden").hide()
+
+$("#mod-license").chosen({width: '100%'}).change((evt, par) ->
+    if (par.selected == "Other")
```
Comment (HebaruSan) [inline frontend/coffee/create.coffee:176]:
```suggestion
    if par.selected == "Other"
```

**Pull request merged**